pkg/daemon/update.go

```diff
@@ -1099,12 +1090,6 @@ func (dn *Daemon) applyExtensions(oldConfig, newConfig *mcfgv1.MachineConfig) er
 // switchKernel updates kernel on host with the kernelType specified in MachineConfig.
 // Right now it supports default (traditional) and realtime kernel
 func (dn *Daemon) switchKernel(oldConfig, newConfig *mcfgv1.MachineConfig) error {
-	// We support Kernel update only on RHCOS nodes
-	if !dn.os.IsRHCOS() {
-		glog.Info("updating kernel on non-RHCOS nodes is not supported")
```

Comment:
This one is actually separate between FCOS and RHCOS because Fedora doesn't ship `kernel-rt` today.  So...I dunno, I guess I'm OK just removing this check anyways, it will just fail in a different way.

Reply:
Oh I missed this one calls `IsRHCOS` instead of `IsCoreOSVariant`. So maybe I should leave that one?

Reply:
I don't have a strong opinion personally.  Leaving it for now would also make sense to keep the commit cleanly targeted to just the `isCoreOS` conditionals.

Reply:
Yeah I think that's cleaner, leaving the check in